Src/IronPython/Runtime/TypecodeOps.cs

```
@@ -9,70 +9,87 @@
+            if (!TryDecomposeTypecode(format.AsSpan(), out byteorder, out code)) {
+                throw PythonOps.ValueError("invalid typecode");
+            }
+        }
+
+        public static bool IsByteCode(char typecode) {
+            return typecode == 'B' || typecode == 'b' || typecode == 'c';
+        }
+
+        public static int GetTypecodeWidth(char typecode) {
+            switch (typecode) {
+                case 'c': // bytechar
                 case 'b': // signed byte
                 case 'B': // unsigned byte
-                    width = 1;
-                    return true;
-                case 'u': // unicode char
+                case '?': // bool
+                    return 1;
                 case 'h': // signed short
                 case 'H': // unsigned short
-                    width = 2;
-                    return true;
+                    return 2;
                 case 'i': // signed int
                 case 'I': // unsigned int
                 case 'l': // signed long
                 case 'L': // unsigned long
                 case 'f': // float
-                    width = 4;
-                    return true;
-                case 'P': // pointer
-                    width = IntPtr.Size;
-                    return true;
+                case 'n': // signed index
+                case 'N': // unsigned index
+                    return 4;
                 case 'q': // signed long long
                 case 'Q': // unsigned long long
                 case 'd': // double
-                    width = 8;
-                    return true;
+                    return 8;
+                case 's': // char pointer
+                case 'p': // char pointer
+                case 'P': // void pointer
+                    return UIntPtr.Size;
                 default:
-                    width = 0;
-                    return false;
+                    return 0;
```

Probably better to throw instead of returning 0?

---

Sure.